File: anthill_front/lib/src/modules/transactions/domain/dtos/transaction_read_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../../../shared/http.dart';
import '../../../users/users_module.dart';

part 'transaction_read_dto.freezed.dart';

part 'transaction_read_dto.g.dart';

@freezed
class TransactionReadDto with _$TransactionReadDto implements IdentifiableModel {
  const factory TransactionReadDto({
    required int id,
    required DateTime createDate,
    DateTime? deleteDate,
    required double amount,
    required bool isIncome,
    required String sourceOrPurpose,
    required String note,
    required UserReadDto user,
  }) = _TransactionReadDto;

  factory TransactionReadDto.fromJson(Map<String, Object?> json) =>
      _$TransactionReadDtoFromJson(json);
}

File: anthill_back/src/modules/transactions/data/dtos/transaction.read.dto.ts
import { IsBoolean, IsDate, IsNotEmpty, IsNumber } from 'class-validator';
import { AutoMap } from 'automapper-classes';
import { UserReadDto } from '../../../users/data/dtos/user.read.dto';

export class TransactionReadDto {
  @AutoMap()
  @IsNumber()
  id: number;

  @AutoMap()
  @IsDate()
  createDate: Date;

  @AutoMap()
  deleteDate?: Date;

  @AutoMap()
  @IsNumber()
  amount: number;

  @AutoMap()
  @IsBoolean()
  isIncome: boolean;

  @AutoMap()
  @IsNotEmpty()
  sourceOrPurpose: string;

  @AutoMap()
  note?: string;

  user: UserReadDto;
}
